cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/RuntimeJobManager.java

```diff
@@ -43,14 +42,6 @@ public interface RuntimeJobManager extends Closeable {
    */
   Optional<RuntimeJobDetail> getDetail(ProgramRunInfo programRunInfo) throws Exception;
 
-  /**
-   * Provides all the jobs that are in running state. If there are no running jobs, it should return empty list.
-   *
-   * @return a list job details
-   * @throws Exception thrown if any exception while getting list of running jobs
-   */
-  List<RuntimeJobDetail> list() throws Exception;
```

Comment:
We can't make backwards incompatible changes in any of the *-api or *-spi modules, so we can't remove this. 

Instead, annotate this as deprecated and make a default implementation that throws an UnsupportedMethodException.

Reply:
Done.

Reply:
this should be a default implementation so that we can remove it from the dataproc and tethering job managers

```
default List<RuntimeJobDetail> list() {
  throw new UnsupportedMethodException(...);
}
```

Reply:
add a default implementation as described above, and remove the overrides in the implementing classes.